src/composables/usePerf.ts

```diff
@@ -0,0 +1,141 @@
+import type { Ref } from 'vue'
+import { provide, reactive, onUnmounted, unref } from 'vue'
+import { useFps, useMemory, useRafFn } from '@vueuse/core'
+import { useTresContext } from '@tresjs/core'
+import type { Scene, WebGLRenderer } from 'three'
+import { calculateMemoryUsage } from '../utils'
+
+interface MemoryUsageData {
+  currentMem: number
+  averageMem: number
+  allocatedMem: number
+  maxMemory: number
+  accumulator: number[]
+  lastLoggedTime: number
+  logInterval: number
+}
+
+interface FpsData {
+  value: number
+  accumulator: number[]
+  lastLoggedTime: number
+  logInterval: number
+}
+
+// Define the structure of the performance data
+interface PerfData {
+  fps: FpsData
+  memory: MemoryUsageData // Adjust 'any' to a more specific type as needed
+  gl: WebGLRenderer | undefined
+  scene: Scene | undefined
+}
+
+// Define the structure of the store
+interface PerfStore {
+  [uuid: string]: { perf: PerfData }
+}
+
+export const PERF_CONTEXT_KEY = Symbol('PERF_CONTEXT_KEY')
+const DEFAULT_UUID = 'default'
+
+// Internal state
+const perfStore = reactive<PerfStore>({
+  [DEFAULT_UUID]: {
+    perf: {
+      fps: reactive({
+        value: 0,
+        accumulator: [],
+        lastLoggedTime: Date.now(),
+        logInterval: 1000,
+      }),
+      memory: reactive({
+        currentMem: 0,
+        averageMem: 0,
+        maxMemory: 0,
+        allocatedMem: 0,
+        accumulator: [],
+        lastLoggedTime: Date.now(),
+        logInterval: 1000,
+      }), // Placeholder for memory object
+      gl: undefined,
+      scene: undefined,
+    },
+  },
+})
+
+export function usePerfProvider(uuid: string = DEFAULT_UUID) {
+  provide(PERF_CONTEXT_KEY, perfStore)
+  return perfStore[uuid].perf
+}
+
+export interface TresPerfOptions {
+  renderer?: Ref<WebGLRenderer>
+}
+
+export function usePerf(options?: TresPerfOptions) {
+  const { renderer: TresRenderer, scene } = useTresContext()
+  const gl = options?.renderer || TresRenderer.value
+  if (!gl) {
+    throw new Error('Renderer not found')
+  }
+
+  const perf = perfStore[DEFAULT_UUID].perf
+  perf.gl = gl as WebGLRenderer
+  perf.scene = scene.value as Scene
+
+  const updateInterval = 100 // Update interval in milliseconds
+  const fps = useFps({ every: updateInterval }) 
+  const { isSupported, memory } = useMemory({ interval: updateInterval })
```

Comment:
`useMemory` isn't supported in all evergreen browsers. Not a big problem in itself, but it would probably be good to display a message in the interface, like on the VueUse site:

<img width="751" alt="Screenshot 2023-12-10 at 18 29 48" src="https://github.com/Tresjs/leches/assets/20469369/4a9db21d-c901-41ca-96e4-53c0dd8b59c8">

Here's what the Leches memory output looks like in current FF (120.0.1) – "0 MB":

<img width="292" alt="Screenshot 2023-12-10 at 18 30 57" src="https://github.com/Tresjs/leches/assets/20469369/cfb566b0-dd97-406a-be10-c2c2b9ce8bdc">